libsolc/libsolc.cpp

```diff
@@ -94,16 +94,46 @@ extern char const* solidity_version() noexcept
 {
 	return VersionString.c_str();
 }
+
+// The strings in this list must not be reiszed after they have been added here (via solidity_alloc()), because
+// this may potentially change the pointer that was passed to the caller from solidity_alloc().
+static list<string> solidity_allocations;
+
 extern char const* solidity_compile(char const* _input, CStyleReadFileCallback _readCallback, void* _readContext) noexcept
 {
-	s_outputBuffer = compile(_input, _readCallback, _readContext);
-	return s_outputBuffer.c_str();
+	return solidity_allocations.emplace_back(compile(_input, _readCallback, _readContext)).data();
 }
-extern void solidity_free() noexcept
+
+extern char* solidity_alloc(size_t _size) noexcept
+{
+	try
+	{
+		solidity_allocations.emplace_back(_size, '\0');
+		return solidity_allocations.back().data();
+	}
+	catch (...)
+	{
+		// most likely a std::bad_alloc(), if at all.
+		return nullptr;
+	}
+}
+
+extern void solidity_free(char* _data) noexcept
+{
+	for (auto i = begin(solidity_allocations); i != end(solidity_allocations); ++i)
```

Comment:
Probably should also enclose all this in a `try/catch`, since we're marking it `noexcept`.

Reply:
Since we agreed to `abort()` here upon invalid argument, we don't need to try/catch here, do we?

Reply:
If we are 100% sure the `noexpect` specifier will cause the compiler to insert exception catching and always abort, then we don't need to do these manually, in any of the functions.